### PR TITLE
Enemy Movement

### PR DESCRIPTION
**Pull request by Shom770**
- [x] Add ``mill`` function for the enemy to move around randomly

### CODE REVIEW

File: src/resources/entities/Enemy.py [hunk 1]
```diff
@@ -1,8 +1,38 @@
...
+        if movement_axis == 'x':  # movement will be on the x axis
+            if (self.x <= 1 and movement == -1) or (self.x >= self.level.width - 2 and movement == 1):
+                direction_axis.remove(movement)
+                self.x += int(direction_axis[0])
+            else:
+                self.x += movement
+        if movement_axis == 'y':  # movement will be on the y axis
+            if (self.y <= 1 and movement == -1) or (self.y >= self.level.height - 2 and movement == 1):
+                direction_axis.remove(movement)
+                self.y += int(direction_axis[0])
```
Comment (n0remac) [inline src/resources/entities/Enemy.py]: Why are you doing `direction_axis.remove(movement)` then `self.y += int(direction_axis[0])`?

I see that it works, but there must be a much cleaner way to do this.

File: src/resources/entities/Enemy.py [hunk 2]
```diff
@@ -1,8 +1,38 @@
...
+            else:
+                self.y += movement
+
+    def draw(self) -> None:
+        """Places entity on map"""
+        self.level.board[self.y][self.x] = self.ground_symbol
+        self.update()
+        self.level.board[self.y][self.x] = self.symbol
```
Comment (n0remac) [inline src/resources/entities/Enemy.py:35]: draw and update should really be separate from each other. Both are suppoesed to be called by the enemy manager.

**Review by n0remac — CHANGES_REQUESTED** [review]

File: src/resources/entities/Enemy.py (hunk 1 above)
Comment (Shom770) [reply src/resources/entities/Enemy.py]: I could probably make it a one liner to
```py
self.y += int(''.join(list(map(str, direction_axis))).replace(str(movement), ''))
```
but that's not readable at all hence I did that

File: src/resources/entities/Enemy.py (hunk 2 above)
Comment (Shom770) [reply src/resources/entities/Enemy.py:35]: Will fix that

File: src/resources/entities/Enemy.py (hunk 1 above)
Comment (n0remac) [reply src/resources/entities/Enemy.py]: It shouldn't be a one liner, but the way it is now it's hard to understand. Can't you just chose and axis then apply the random choose of direction to the access without mutating a list?

Comment (Shom770) [reply src/resources/entities/Enemy.py]: If the direction leads to the edge that's why I had to mutate the list, I can do another way though

Comment (Shom770) [reply src/resources/entities/Enemy.py]: Resolved, I overthought it when I could just do a simple ternary

**Pull request merged**